webrtc.html

```diff
@@ -9712,7 +9712,41 @@ <h2>Attributes</h2>
             <dt><dfn data-idl><code>canInsertDTMF</code></dfn> of type <span class=
             "idlAttrType"><a>boolean</a></span>, readonly</dt>
             <dd>
-              <p>Whether the <a>RTCDTMFSender</a> is capable of sending DTMF.</p>
+              <p>Whether the <a>RTCDTMFSender</a> <var>dtmfSender</var> is capable of
+              sending DTMF. On getting, the attribute MUST return the value of
+              <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot. To set
+              <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot, run the following steps
+              when <code><var>pc</var>.onsignalingstatechange</code> or
+              <code><var>pc</var>.connectionstatechange</code> is fired:
+              <ol>
+                <li>Let <var>pc</var> be the <code><a>RTCPeerConnection</a></code> which
+                is sending DTMF.</li>
+                <li>Let <var>transceiver</var> be the <code><a>RTCRtpTransceiver</a></code>
+                used to send DTMF.</li>
+                <li>Check whether <code><var>pc</var>.signalingState</code> is <code>"stable"</code>.
+                If not, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to <code>false</code>
+                and abort these steps.</li>
+                <li>Check whether <code><var>pc</var>.connectionState</code> is <code>"connected"</code>.
+                If not, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to <code>false</code>
+                and abort these steps.</li>
+                <li>Check whether <code><var>transceiver</var>.sender.track</code> is set.
+                If not, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to <code>false</code>
+                and abort these steps.</li>
+                <li>Check whether <code><var>transceiver</var>sender.track.kind</code> has the value
+                <code>"audio"</code>. If not, set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to
+                <code>false</code> and abort these steps.</li>
+                <li>Check whether <code><var>transceiver</var>.currentDirection</code> is set. If not,
+                set <var>dtmfSender</var>'s <a>[[\CanInsertDTMF]]</a> slot to <code>false</code>
+                and abort these steps.</li>
+                <li>Check whether <code><var>transceiver</var>.currentDirection</code> has the value
+                <code>"sendrecv"</code> or <code>"sendonly"</code>. If not, set <var>dtmfSender</var>'s
+                <a>[[\CanInsertDTMF]]</a> slot to <code>false</code> and abort these steps.</li>
+                <li>Check whether for some value of <var>j</var> ranging from 0 to the number of codecs,
+                <code><var>transceiver</var>.sender.getParameters().codecs[<var>j</var>].mimetype</code> has the value
```

Comment:
does the clockRate need to match a negotiated codec too? Chrome has
```
a=rtpmap:110 telephone-event/48000
a=rtpmap:112 telephone-event/32000
a=rtpmap:113 telephone-event/16000
a=rtpmap:126 telephone-event/8000
```
(and maybe @alvestrand knows why)

Reply:
Yes, it does need to match a negotiated codec clock rate, so that DTMF can be sent with the same SSRC as that other codec (clockrate can't vary on an SSRC).  PR assumes that the SDP negotiation handles that correctly.

Reply:
and .codecs is only set after negotiation. Now it makes sense, thank you!